networkdb/delegate.go

```diff
@@ -140,7 +140,7 @@ func (nDB *NetworkDB) handleNetworkEvent(nEvent *NetworkEvent) bool {
 	nDB.Lock()
 	defer nDB.Unlock()
 
-	if nEvent.NodeName == nDB.config.NodeName {
+	if nEvent.NodeName == nDB.config.NodeID {
```

Comment:
Especially such changes where previously learnt messages from a node (prior to upgrading with this change) would have used NodeName in the messages. After the upgraded software, this node will be using NodeID, while the previous states maintained by the peers will still have the older states with the nodeName. And this check will fail, which is not expected.